### PR TITLE
Integration test for pilot DE process (SCP-4204)

### PR DESCRIPTION
**Pull request by jlchang**
Using a small test dataset derived from public study SCP1677, implemented an integration test for the pilot DE pipeline. This test runs the equivalent of:

> ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed0000000000000 differential_expression --annotation cell_type__ontology_label --matrix-file-path ../tests/data/differential_expression/de_integration.tsv --matrix-file-type dense --cell-metadata-file ../tests/data/differential_expression/de_integration_unordered_metadata.tsv --cluster-file ../tests/data/differential_expression/de_integration_cluster.tsv --name de_integration --study-accession SCPde --differential_expression 

and checks that expected output file "de_integration--cell_type__ontology_label--cholinergic_neuron--wilcoxon.tsv"
has the same md5 checksum as the checksum calculated using the reference file in `tests/data/differential_expression/reference`

To test locally, in the test directory, run:
pytest -k test_de_process
and confirm that it passes.

This satisfies SCP-4204.

### CODE REVIEW

File: ingest/de.py [hunk 1]
```diff
@@ -171,6 +171,12 @@ def execute_de(self):
                 self.method,
             )
             DifferentialExpression.de_logger.info("preparing DE on dense matrix")
+        else:
+            msg = f"Submitted matrix_file_type should be 'dense' or 'mtx' not {self.matrix_file_type}"
```
Comment (eweitz) [inline ingest/de.py:175]:
```suggestion
            msg = f"Submitted matrix_file_type should be \"dense\" or \"mtx\" not \"{self.matrix_file_type}\""
```

**Review by eweitz — APPROVED** [review]

**Review by devonbush — CHANGES_REQUESTED** [review]
As we talked about, let's add a check for the number of files generated, the filenames, and a spot check of some values

File: ingest/de.py (hunk 1 above)
Comment (jlchang) [reply ingest/de.py]: `7205b9b`

**Comment by jlchang** [comment]
Added more assertions to `test_de_process` to check presence of expected output files, file naming and spot-check of file content.

File: tests/test_de.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+def find_expected_files(labels, cluster_name, annotation, method):
+    """ Check that files were created for all expected annotation labels
+    """
+    found = 0
+    for label in labels:
+        sanitized_label = label.replace(" ", "_")
+        expected_file = (
+            f"{cluster_name}--{annotation}--{str(sanitized_label)}--{method}.tsv"
```
Comment (eweitz) [inline tests/test_de.py:36]: `str` seems extraneous; `replace` assumes string, and f-string interpolation coerces to string.

```suggestion
        expected_file = (
            f"{cluster_name}--{annotation}--{sanitized_label}--{method}.tsv"
```

File: ingest/de.py [hunk 3]
```diff
@@ -171,6 +171,12 @@ def execute_de(self):
                 self.method,
             )
             DifferentialExpression.de_logger.info("preparing DE on dense matrix")
+        else:
+            msg = f"Submitted matrix_file_type should be \"dense\" or \"mtx\" not {self.matrix_file_type}"
```
Comment (eweitz) [inline ingest/de.py:175]: Slightly easier readability:

```suggestion
            msg = f"Submitted matrix_file_type should be \"dense\" or \"mtx\" not \"{self.matrix_file_type}\""
```

**Review by eweitz — APPROVED** [review]
Looks good!  I noted some non-blocking nits.

File: tests/test_de.py (hunk 2 above)
Comment (jlchang) [reply tests/test_de.py:36]: Good point. The string coercion was put in place before I started sanitizing the label so any coercion is needed before the sanitizing step. So I need to step back and think about what is needed if a cell is not annotated with cell_type__ontology_label (cell_type is optional metadata so this is reasonable in our validated metadata). For those cases the label would be NaN which is a float (and would be represented in visualizations as "Unspecified").

**Review by devonbush — APPROVED** [review]
Nice, appreciate the spot-checks

File: tests/test_de.py (hunk 2 above)
Comment (jlchang) [reply tests/test_de.py:36]: I work on addressing the above issue in`jlc_sparse_de` to avoid more "artifactual merge commits"

**Pull request merged**